src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -23,8 +23,77 @@ namespace System
     /// </summary>
     public static class Span
     {
-        // s_invariantMode is defined for the perf reason as accessing the instance field is faster than access the static property GlobalizationMode.Invariant
-        private static readonly bool s_invariantMode = GlobalizationMode.Invariant;
+        /// <summary>
+        /// Copies the characters from the source span into the destination, converting each character to lowercase.
+        /// </summary>
+        public static int ToLower(this ReadOnlySpan<char> source, Span<char> destination, CultureInfo culture)
+        {
+            if (culture == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.culture);
+
+            // Assuming that changing case does not affect length
+            if (destination.Length < source.Length)
+                return -1;
+
+            if (GlobalizationMode.Invariant)
+                culture.TextInfo.ToLowerAsciiInvariant(source, destination);
+            else
+                culture.TextInfo.ChangeCase(source, destination, toUpper: false);
+            return source.Length;
+        }
+
+        /// <summary>
+        /// Copies the characters from the source span into the destination, converting each character to lowercase
+        /// using the casing rules of the invariant culture.
+        /// </summary>
+        public static int ToLowerInvariant(this ReadOnlySpan<char> source, Span<char> destination)
+        {
+            // Assuming that changing case does not affect length
+            if (destination.Length < source.Length)
+                return -1;
+
+            if (GlobalizationMode.Invariant)
+                CultureInfo.InvariantCulture.TextInfo.ToLowerAsciiInvariant(source, destination);
+            else
+                CultureInfo.InvariantCulture.TextInfo.ChangeCase(source, destination, toUpper: false);
```

Comment:
@tarekgh, is there a difference between these two operations?

Reply:
Yes, there is a difference. ToLowerAsciiInvariant is just trying to be optimized for the Ascii but ChangeCase is not. for example, if you have a Turkish text would be handled wrong if you do it with ToLowerAsciiInvariant

Reply:
Isn't this naming confusing then?  Turning on the "Invariant" mode potentially produces different results than using the InvariantCulture?

Reply:
well, Invariant mode means we'll use the Invariant culture data all the time (without going to call the underlying OS) this will include doing all sorting/casing operations as ordinal. I believe whoever introduced ToLowerAsciiInvariant meant to have this operation mainly for Ascii strings and at the same time not using a culture like Turkish to avoid the Turkish 'I' problem. but in general, you are right for the statement, the invariant mode can produce different results than invariant culture in the sorting and casing area.

Reply:
Ok. Thanks.